pkg/broker/file.go

```diff
@@ -92,7 +92,7 @@ func ReadInfoFromFile(filename string) (*Info, error) {
 	return data, errors.Wrap(json.Unmarshal(bytes, data), "error unmarshalling data")
 }
 
-func newDataFrom(kubeClient kubernetes.Interface, brokerNamespace, ipsecFile string) (*Info, error) {
+func newDataFrom(kubeClient kubernetes.Interface, brokerNamespace string, ipsecPSK []byte) (*Info, error) {
```

Comment:
Not critical to the change, but since you're already touching it you could refactor the function to receive the `restConfig` and create `kubeClient` inside the function, as it's the only one using it.

Looking at it a bit further, the whole function is silly and could be broken down into "read client token" and "initialize PSK" which both can be called from `WriteInfoToFile`.

Reply:
> Not critical to the change, but since you're already touching it you could refactor the function to receive the `restConfig` and create `kubeClient` inside the function, as it's the only one using it.

Keeping it as `kubeClient` will make it easier to add unit tests in the future.

Reply:
As I wrote this function is quite silly and can just be removed overall, even more so if we agree on https://github.com/submariner-io/subctl/pull/588/files#r1129189996

Let's see what @skitt thinks